must-gather/collection-scripts/gather_versions

```diff
@@ -0,0 +1,81 @@
+function log_command() {
+    heading=$1
+    command=$2
+    echo "***" >> ${debug_info}
+    echo "* $heading" >> ${debug_info}
+    $command 2>&1 >> ${debug_info}
+    echo -e "\n\n" >> ${debug_info}
+}
+
+# Search for given operator in all available namespaces, and save information
+function log_operator() {
+    display_name=$1
+    operator_name=$2
+
+    # Find all operators with matching name in all namespaces
+    namespaces=$(oc get subscriptions.operators.coreos.com --all-namespaces --insecure-skip-tls-verify=${skip_tls} \
+                -o jsonpath='{.items[?(.metadata.name=="'"${operator_name}"'")].metadata.namespace}')
+    if [ -z "${namespaces}" ]; then # No matches, mark as not found
+        log_command "${display_name} information" "echo ${operator_name} operator not found"
+    else # Loop through matching operators and save information to debug file
+        for namespace in ${namespaces}; do
+            log_command "${display_name} version information in namespace ${namespace}" \
+                        "oc get subscriptions.operators.coreos.com ${operator_name} -n ${namespace} --insecure-skip-tls-verify=${skip_tls} -o yaml"
+        done
+    fi
+}
+
+# Search for named deployments in all namespaces
+function log_deployment() {
+    display_name=$1
+    deployment_name=$2
+
+    namespaces=$(oc get deployment --all-namespaces --insecure-skip-tls-verify=${skip_tls} \
+                 -o jsonpath='{.items[?(.metadata.name=="'"${deployment_name}"'")].metadata.namespace}')
+    if [ -z "${namespaces}" ]; then
+        log_command "${display_name} information" "echo ${deployment_name} deployment not found"
+    else
+        for namespace in ${namespaces}; do
+            log_command "${display_name} deployment information in namespace ${namespace}" \
+                        "oc get deployment ${deployment_name} -n ${namespace} --insecure-skip-tls-verify=${skip_tls} -o yaml"
+        done
+    fi
+}
+
+# OpenShift version
+log_command "OpenShift version information"\
+            "oc version --insecure-skip-tls-verify=${skip_tls} -o yaml"
+
+# Red Hat OADP operator information
+log_operator "OADP" "redhat-oadp-operator"
+
+# Community OADP operator
+log_operator "Community OADP operator" "oadp-operator"
+
+# Volsync operator information
+log_operator "Volsync" "volsync-product"
+
+# Restic version
+if [ -z "${skip_tls}" ]; then
+    log_command "Restic version" "restic version"
+else
+    log_command "Restic version" "restic version --insecure-tls"
+fi
+
+# Velero versions
+log_command "Velero client version" "velero version --client-only"
+log_deployment "Velero" "velero"
+
+# Storage classes
+log_command "Storage classes" "oc get storageclass --all-namespaces --insecure-skip-tls-verify=${skip_tls}"
+
+# DataProtectionApplications
+log_command "DataProtectionApplication CRs" "oc get dpa --all-namespaces --insecure-skip-tls-verify=${skip_tls}"
```

Comment:
same thing here.. we need the full yaml output of all dpa's

Reply:
I did this before and took it out because I noticed [gather_crs](https://github.com/openshift/oadp-operator/blob/master/must-gather/collection-scripts/gather_crs) already does this. Is it still worth copying these whole YAML blobs into the one top-level oadp-debug-info file, versus showing a more targeted summary?

Reply:
it's probably easier to recreate them reuse what's already generated.  The point is to have the very important information in one file.   Having it in one file is my only concern, how it gets there is up to you

Reply:
I prefer an organized folder of separate files when receiving must gather info.. but hey you do you.

Reply:
To minimize size, perhaps a script can be made to create the one file from an already gathered resources to be run during debugging.. to keep mustgather small

Reply:
Customers have had timeout issues with large clusters running must gather before.. and they will again. Need to minimize performance impact during gathering.

Reply:
+1

Reply:
I found plenty of ways to improve performance, but I started to get bogged down in options. I think it would be better to get this stuff in the way it is now, and then I can add more to #1002 for OADP-148 performance improvements, since that's pretty much what it's already there for.

Reply:
I agree w/ @mrnold